### PR TITLE
fix(flaps/spoilers): Fix flap animation position and interim fix for spoilers animation low end too smooth 

### LINKED ISSUE #9426: [A380X] 5° spoiler raise when on ground in clean or moving
### Aircraft Version

Stable

### Description

Add special ground spoiler logic. The A380 spoilers do a 5° raise when on the ground with at least 1 engine started, in clean configuration, or flaps are in the process of extending or retracting from CONF 1 to CONF 2. 

### References (optional)

(Timestamps included where necessary)

https://youtu.be/wBCU2aEjYno?si=070L_75C6N0AT1TI&t=158
https://youtu.be/PrNz5-2Q72c?si=giLkjLDtM92liZ1_ (in this video, the middle 3 spoilers can be seen lowering back before the inner and outer spoilers as flaps transition to CONF 2)
https://youtu.be/yB2ES1jWlow?si=NTNGdT_-i5u66sCA&t=1057
https://www.pprune.org/tech-log/519217-a380-spoilers-movement.html

### Additional info (optional)

I don't have access to the FCOM but it could possibly say something about this spoiler logic in there somewhere, if anyone who does have access wants to check.

### Discord Username (optional)

hidinginabush

### PR DESCRIPTION
**Pull request by crocket63**
Fixes #9426

## Summary of Changes

Flaps animation now driven from the FPPU curve instead of linearly from min to max angle.
Spoiler animation is wrongly smoothed in the model. While waiting for a model fix there's an added gain on animation low end so at 5° deployment it looks okish.

## References

https://www.youtube.com/watch?v=brIwF5_1xdA

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Check Flaps animation position is working and at correct position relative to real references
Both side work
Spoiler panels should all work normally on both side, and visually deploy on ground at 5° when between flap 0 and flap 2.

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -27,8 +27,10 @@
 1. [A380X/AP] Improved support of simulation rate 4x - @aguther (Andreas Guther)
 1. [A32NX/FANS] Restore Portuguese station codes - @tracernz (Mike)
 1. [A380X/MFD] Fixed DIRECT TO selection of active waypoint does nothing - @sognodelx (Sven Gross)
-1. [A380X/FLIGHT MODEL] Fix for stall and auto-rotation issues- @donstim (donbikes)
+1. [A380X/FLIGHT_MODEL] Fix for stall and auto-rotation issues- @donstim (donbikes)
 1. [A380X/MODEL] Optimized 4K textures - @Repsol
+1. [A380/ANIM] Animation of flaps now from FPPU position. Interim fix for spoiler low end animation - @Crocket63 (crocket)
+
```
Comment (2hwk) [inline .github/CHANGELOG.md:33]:
```suggestion
```

Unnecessary new line here

**Review by 2hwk — DISMISSED** [review]

**Comment by 2hwk** [comment]
Rust lint is also failing.

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -27,8 +27,9 @@
 1. [A380X/AP] Improved support of simulation rate 4x - @aguther (Andreas Guther)
 1. [A32NX/FANS] Restore Portuguese station codes - @tracernz (Mike)
 1. [A380X/MFD] Fixed DIRECT TO selection of active waypoint does nothing - @sognodelx (Sven Gross)
-1. [A380X/FLIGHT MODEL] Fix for stall and auto-rotation issues- @donstim (donbikes)
+1. [A380X/FLIGHT_MODEL] Fix for stall and auto-rotation issues- @donstim (donbikes)
```
Comment (beheh) [inline .github/CHANGELOG.md:30]: I don't think this PR should change unrelated changelog lines.

File: fbw-a380x/src/wasm/systems/a380_systems_wasm/src/spoilers.rs [hunk 3]
```diff
@@ -35,5 +35,111 @@ pub(super) fn spoilers(builder: &mut MsfsAspectBuilder) -> Result<(), Box<dyn Er
         Variable::named("HYD_SPOILERS_RIGHT_DEFLECTION"),
     );
 
+    // FIX ME: Remove this section once spoiler animation is back to linear in the model
+    builder.map(
```
Comment (Gurgel100) [inline fbw-a380x/src/wasm/systems/a380_systems_wasm/src/spoilers.rs:39]: Shouldn't these be `ExecuteOn::PostTick`? Otherwise you always use the values from the previous frame if I understand the aspect layer correctly.

File: fbw-common/src/wasm/systems/systems/src/hydraulic/flap_slat.rs [hunk 4]
```diff
@@ -532,6 +539,20 @@ impl SimulationElement for FlapSlatAssembly {
...
+                / self.max_synchro_gear_position.get::<degree>()
+                * 100.,
+        );
+
+        writer.write(
+            &self.animation_right_id,
+            self.position_feedback().get::<degree>()
+                / self.max_synchro_gear_position.get::<degree>()
+                * 100.,
+        );
```
Comment (Gurgel100) [inline fbw-common/src/wasm/systems/systems/src/hydraulic/flap_slat.rs:554]:
```suggestion
        writer.write(
            &self.animation_left_id,
            (self.position_feedback() / self.max_synchro_gear_position).get::<percent>(),
        );

        writer.write(
            &self.animation_right_id,
            (self.position_feedback() / self.max_synchro_gear_position).get::<percent>(),
        );
```

**Review by Gurgel100 — APPROVED** [review]

**Review by 2hwk — APPROVED** [review]

**Comment by 2hwk** [comment]
Quality Assurance Report

Discord Username: 2cas
Object of testing: #9445
Aircraft: A380X 
Tier of Testing: 1
Date: 10/11/2024 

Testing Process: 
 
- [x] Check Flaps animation position is working and at correct position relative to real references
- [x] Both side work
- [x] Spoiler panels should all work normally on both side, and visually deploy on ground at 5° when between flap 0 and flap 2.
Testing Results: 
QA Passed  

Negatives:       

- Doesn't exactly match the source videos for now, as the 5 degree spoiler deployment is a bit before the flap actuators start moving, but that's not in the scope of this PR.

Conclusions: 

- LGTM, was a fun PR to test

**Comment by C-Schaffhausen** [comment]
Flew with the master build including the spoiler fix. After starting Eng 1+2, a few spoilers raised to 5 degrees already, while the video here shows that the spoilers raise at the same time after engine start. Could you have a look? Thank you!

https://youtu.be/lrxxsPryoqY?si=jnj2aSJ2Q5bKZ4vI&t=664
